src/tls_stream.rs

```diff
@@ -759,9 +762,6 @@ impl<S> TlsStream<S>
 
             match secur32::EncryptMessage(self.context.get_mut(), 0, &mut bufdesc, 0) {
                 winapi::SEC_E_OK => {
-                    let len = bufs[0].cbBuffer + bufs[1].cbBuffer + bufs[2].cbBuffer;
-                    self.out_buf.get_mut().truncate(len as usize);
```

Comment:
Why are you ignoring the actual size of the encrypted data?
I don't see where else the out_buf is restricted to it, this could lead to invalid writes.